### ISSUE TITLE
How many epochs to take during a compression process

### ISSUE THREAD

**Issue by as754770178** [Tencent/PocketFlow#88]
How to calculate how many epochs to take during a compression process.
That is `nb_iters_train` * `how many combinations of hyper parameters` * `how many bucket divided in one model` ?
The parameter `nb_iters_train` is calculated by the function `setup_lrn_rate` in `lrn_rate_util.py`.

**Comment by as754770178** [comment]
Such as UniformQuantTFLearner and ChannelPrunedLearner.

**Comment by ginger0106** [comment]
I tested the demo ` ./scripts/run_docker.sh nets/resnet_at_ilsvrc12_run.py -n=4 \
    --learner dis-chn-pruned` , the idx_iter is 339800 after one day using 4 GPUs, I wonder when it would be completed

**Comment by ginger0106** [comment]
@jiaxiang-wu  Got it, Thanks.

**Comment by as754770178** [comment]
what the number `1281167 ` mean?

**Comment by as754770178** [comment]
Can I understand that `nb_epochs` mean the total epochs in entire compression process, so that the sum epoch numbers of every bucket and parameter combiantion.  It's not epoch numbers of one bucket for one parameter combinaition.

**Comment by jiaxiang-wu** [comment]
@as754770178 What do you mean by "bucket"? Which model compression component are you using?

**Comment by as754770178** [comment]
bucket mean that `uql_use_buckets` in UniformQuantLearner.  With bucketing, weights are split into multiple pieces.  model compression component such as UniformQuantLearner.

**Comment by jiaxiang-wu** [comment]
@haolibai Could you help to explain the number of epochs required in `UniformQuantLearner`?

**Comment by haolibai** [comment]
In `UniformQuantLearner`, the number of epochs for fine-tuning is calculated by the function `setup_bnds_decay_rates`. Following a similar pattern as the following, suppose the default number for training is 60 on Cifar-10, the batch size per GPU is 128, then the iteration will be:
60000 * 60 / (128*4) = 7031.

> For ResNet models, the default number of training epochs is 100. The batch size per GPU is 64, and since you are using 4 GPUs, the number of training iterations is given by:
> 1281167 * 100 / (64 * 4) = 500456
> @ginger0106

What do you mean by `nb_iters_train` * `how many combinations of hyper parameters` * `how many bucket divided in one model`? The number of buckets does not affect the epochs of training. For different combination of parameters, the RL agent can automatically optimize them. For example, `uql_tune_global_steps` determines how many steps for training the model to get the reward for the current combination of hyper-parameters.
